module/zfs/zvol.c

```diff
@@ -1531,23 +1507,78 @@ zvol_compat_ioctl(struct block_device *bdev, fmode_t mode,
 #define	zvol_compat_ioctl	NULL
 #endif
 
+/*
+ * Linux 2.6.38 preferred interface.
+ */
+#ifdef HAVE_BLOCK_DEVICE_OPERATIONS_CHECK_EVENTS
+static unsigned int
+zvol_check_events(struct gendisk *disk, unsigned int clearing)
+{
+	unsigned int mask = 0;
+
+	rw_enter(&zvol_state_lock, RW_READER);
+
+	zvol_state_t *zv = disk->private_data;
+	if (zv != NULL) {
+		boolean_t locked = MUTEX_HELD(&zv->zv_state_lock);
```

Comment:
For this function, as well as zvol_media_changed() and zvol_revalidate_disk(), I am not sure why we don't know if we are holding the zv_state_lock(). Can some comments be added here as to the possible locking states of the calling thread and the context for those states to take place?

Reply:
Good idea, will do.